exchangelib/util.py

```diff
@@ -512,7 +512,7 @@ def post_ratelimited(protocol, session, url, headers, data, allow_redirects=Fals
             # Always create a dummy response for logging purposes, in case we fail in the following
             r = DummyResponse(url=url, headers={}, request_headers=headers)
             try:
-                data = data.encode('utf-8')
+                data = ensure_binary(data)
```

Comment:
I think I can also remove lines L515 to L522 since we shouldn't enter the `UnicodeDecodeError` due to `ensure_binary`, but I'm not 100% sure so I left out that change.

Reply:
I'm not clear on what exactly that error handling is trying to achieve. @yusra-ahmed it looks like you added to this error handling logic about a year ago. If you happen to have additional context here, that'd be helpful.

Reply:
This change was a result of a super gnarly bug where events weren't syncing back due to unicode characters in the description...if I'm remembering correctly. Went through several iterations and landed on this fix that alleviated the issue. Might be worth testing that we're not breaking events syncback if the description has unicode characters.
More context here: https://app.clubhouse.io/nylas/story/27965/unicodedecodeerror-in-syncback-persisting-at-low-er-rate-when-certain-unicode-characters-are-used-in-the-event-payload

Reply:
Addressed by adding tests to cloud-core (https://github.com/nylas/cloud-core/pull/4667#issuecomment-908610203)